carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/listener/DriverListener.java

```diff
@@ -162,6 +162,20 @@ public void onException(Throwable thr, WebDriver driver) {
 				return;
 			}
 			
+			// handle use-case when application crashed on iOS but tests continue to execute something because doesn't raise valid exception
+			// Example:
+
+			// 10:25:20 2018-09-14 10:29:39 DriverListener [TestNG-31] [ERROR]
+			// [iPhone_6s] An unknown server-side error occurred while
+			// processing the command. Original error: The application under
+			// test with bundle id 'Q5AWL8WCY6.iMapMyRun' is not running,
+			// possibly crashed (WARNING: The server did not provide any
+			// stacktrace information)
+			
+			if (thr.getMessage().contains("is not running, possibly crashed")) {
+				throw new RuntimeException(thr);
+			}
+			
 			
 			// handle cases which should't be captured
 			if (!thr.getMessage().contains("StaleObjectException")
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (10) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://demo.qaprosoft.com/sonarqube/coding_rules#rule_key=squid%3AS1067)